test/Editor.cy.tsx

```diff
@@ -222,6 +223,53 @@ describe("<Editor />", () => {
       )
     })
   })
+
+  describe("selection state", () => {
+    it("AllSelection kept on adding mark", () => {
+      const handle = makeHandle({ text: "Hello Happy World" })
+      mount(
+        <Editor
+          handle={handle}
+          path={["text"]}
+          schemaAdapter={basicSchemaAdapter}
+        />,
+      )
+      editorContents().should("have.html", expectedHtml(["Hello Happy World"]))
+
+      const isMac = Cypress.platform === "darwin"
+
+      const selectAllKey = isMac ? "{cmd}a" : "{ctrl}a"
+      cy.get("div#editor div[contenteditable=true]").type(selectAllKey, {
+        release: false,
+      })
+
+      cy.get("div#editor").then($el => {
+        const view = ($el[0] as { view: EditorView }).view
+        if (!view) {
+          throw new Error("Editor view not found")
+        }
+        expect(view.state.selection.toJSON().type).to.eq("all")
```

Comment:
Is there a way we can test this without using the prosemirror API? Ideally we would just use the browser APIs to determine what area is selected and check that it matches what we expect. That may be gratuitously difficult to do though in which case this is fine.

Reply:
I would defer to ProseMirror here as the test basically checks if AutoMerge doesn’t affect selection; and ProseMirror normalizes selection behavior across browsers. Things like AllSelection, NodeSelection, CellSelection, etc don’t exist outside of ProseMirror (?).